pypit/calibrations.py

```diff
@@ -0,0 +1,512 @@
+
+    def _reset_internals(self):
+        self.msarc = None
+        self.msbias = None
+        self.mbpm = None
+        self.pixlocn = None
+        self.tslits_dict = None
+        self.maskslits = None
+        self.wavecalib = None
+        self.mstilts = None
+        self.mspixflatnrm = None
+        self.slitprof = None
+        self.mswave = None
+        self.datasec_img = None
+
+    def reset(self, setup, det, sci_ID, settings):
+        self.setup = setup
+        self.det = det
+        self.sci_ID = sci_ID
+        self.settings = settings.copy()
+
+        # Setup the calib_dict
+        if self.setup not in self.calib_dict.keys():
+            self.calib_dict[self.setup] = {}
+
+        # Reset internals to None
+        self._reset_internals()
+
+    def get_arc(self, bias=None):
+        # Checks
+        if bias is not None:
+            self.msbias = bias
+        if self.msbias is None:
+            msgs.error("msbias needs to be set prior to arc")
+        # Check
+        self._chk_set(['setup', 'det', 'sci_ID', 'settings'])
+        #
+        if 'arc' in self.calib_dict[self.setup].keys():
+            self.msarc = self.calib_dict[self.setup]['arc']
+        else:
+            # Instantiate with everything needed to generate the image (in case we do)
+            self.arcImage = arcimage.ArcImage([], spectrograph=self.fitstbl['instrume'][0],
+                                settings=self.settings, det=self.det, setup=self.setup,
+                                sci_ID=self.sci_ID, msbias=self.msbias, fitstbl=self.fitstbl)
+            # Load the MasterFrame (if it exists and is desired)?
+            self.msarc = self.arcImage.master()
+            if self.msarc is None:  # Otherwise build it
+                msgs.info("Preparing a master {0:s} frame".format(self.arcImage.frametype))
+                self.msarc = self.arcImage.build_image()
+                # Save to Masters
+                if self.save_masters:
+                    self.arcImage.save_master(self.msarc, raw_files=self.arcImage.file_list, steps=self.arcImage.steps)
+            # Return
+            return self.msarc
+            '''
+            # Grab it -- msarc will be a 2D image
+            self.msarc, self.arcImage = arcimage.get_msarc(self.det, self.setup, self.sci_ID,
+                                          self.fitstbl, self.settings, self.msbias)
+            '''
+            # Save
+            self.calib_dict[self.setup]['arc'] = self.msarc
+        # Return
+        return self.msarc
+
+    def get_bias(self):
+        # Checks
+        self._chk_set(['setup', 'det', 'sci_ID', 'settings'])
+        #
+        if 'bias' in self.calib_dict[self.setup].keys():
+            self.msbias = self.calib_dict[self.setup]['bias']
+            msgs.info("Reloading the bias from the internal dict")
+        else:
+            # Init
+            self.biasFrame = biasframe.BiasFrame(settings=self.settings, setup=self.setup,
```

Comment:
Again, I don't see the need for this BiasFrame class. This is just a silly wrapper for the ProcessImages class.  Am I really the only one that seems like this is overkill?

Here are the steps: 

1) istantiate the class with no arguments. 
2) if no master exists call the function build_image. This function takes no arguments, and instantiation takes no arguments, so god only knows what we are operating on here. 
3)  build_image is actually just a wrapper, for process_bias, which also takes no arguments.
4) process_bias is actually just a wrapper which calls process, also no arguments. 
5) Now I have to go look at the ProcessImages class,find the function process and see what it does. 

So what usually would be 5 lines of code, i.e. check if the master is there, if not create a bias frame from an image stack, is now like 50 lines of codes, involving a class and two wrappers. What did we gain from all this complexity?

This example is easy enough to follow since the operation (creating a bias) is trivial. But following this coding model with unnecessary classes, global variables everywhere, and multiple wrappers (and duplicated code) just becomes a nightmare when we are dealing with more complicated tasks. 

The code becomes unreadable, and tedious to debug because 5 routines are being touched. I am not the only one that feels this way. Everyone in my group that has tried to dig into PYPIT finds the code flow and organization very frustrating.